Parsers/ASimAuthentication/Parsers/vimAuthenticationGithub.yaml

```diff
@@ -0,0 +1,134 @@
+  - Name: actor_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: targetappname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: srchostname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventtype_in
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresultdetails_in
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+     let parser = (
+     starttime:datetime=datetime(null), 
+     endtime:datetime=datetime(null), 
+     username_has_any:dynamic = dynamic([]),
+     targetappname_has_any:dynamic = dynamic([]),
+     srcipaddr_has_any_prefix:dynamic = dynamic([]),
+     srchostname_has_any:dynamic = dynamic([]),
+     eventtype_in:dynamic =dynamic([]),
+     eventresultdetails_in:dynamic = dynamic([]),
+     eventresult:string = '*',
+     disabled:bool=false
+     )
+     {
+     let EventResultDetailslookup=datatable(EventOriginalResultDetails:string,EventResult:string,EventType:string,EventSeverity:string )
+     ["user.login","Success","Logon","Informational",
+     "user.logout","Success","Logoff","Informational",
+     "user.two_factor_challenge_success","Success","Logon","Informational",
+     "user.two_factor_challenge_failure","Failure","Logon","Informational",
+     "user.two_factor_requested","Success","Logon","Informational",
+     "user.failed_login","Failure","Logon","Informational",
+     "user.sign_in_from_unrecognized_device_and_location","Success","Logon","High",
+     "user.sign_in_from_unrecognized_device","Success","Logon","High",
+     "user.two_factor_recovery_codes_downloaded","Success","Logon","Informational",
+     "user.new_device_used","Success","Logon","Informational",
+     "user.promote","Success","Elevate","Informational",
+     ];
+     let GithubAuthenticationEvent=EventResultDetailslookup 
+     |project EventOriginalResultDetails
+     | summarize make_set(EventOriginalResultDetails);  
+     GitHubAuditLogPolling_CL
+     | where not(disabled)
+          and action_s in (GithubAuthenticationEvent)
+     | extend TimeGenerated = todatetime(tostring(split(TimeGenerated, '.', 0)[0]))      
+     | where operation_type_s =='authentication'
+          and(isnull(starttime) or TimeGenerated >= starttime) 
+          and (isnull(endtime) or TimeGenerated <= endtime)          
+          and ((array_length(targetappname_has_any) == 0) or ('Github' has_any (targetappname_has_any)))
+          and ((array_length(srchostname_has_any) == 0)) or ((array_length(username_has_any) == 0) or (TargetUsername has_any (username_has_any)))
+          and ((array_length(srcipaddr_has_any_prefix) == 0)) or (SrcIpAddr has_any (srcipaddr_has_any_prefix)) //SrcHostname not available in source or //username,srcIp addr, Eventtype and eventresult done down in the parser  
```

Comment:
Please keep in separate lines. Correct grammar
// SrcHostname not available in source
// srcipaddr_has_any_prefix, eventtype_in and eventresult filtering done later in the parser